### DISCUSSION TITLE
I need help please...

### DISCUSSION THREAD

**Discussion by johnny-dhn** [mu-editor/mu#2246] [Q&A] [answered]
Hello, i would like to know how we do for import in the form of a module (`from [...] import*`) a library Github. Sorry, my english is not good...

More precisely, my problem is that I try to import the github library: https://github.com/waveshare/WSLCD1in8 to run an LCD. It’s a ST7735S by Waveshare for microbit.

But I’m trying to import it into my program. But I don’t know how to do it. Yes, I’m a noob

Thanks for all.

**Comment by dsssssssss9** [comment, marked as answer]
The waveshare library you mention is designed for Makecode programming environment so won't work in  micropython .

Unless someone else has written a suitable micropython library for this display , you may have to change the display to another device that has a library  or see if your project can be made using Makecode ?

**Comment by carlosperate** [comment]
As @dsssssssss9 has mentioned that library is for MakeCode, that while has a Python mode, it is not the same as MicroPython.

If you are up for the challenge you can try to port an existing MicroPython library to the micro:bit, like this one (there might be more in GitHub): https://github.com/boochow/MicroPython-ST7735

Things like `machine.Pin()` and `machine.SPI()` would have to be translated to the micro:bit versions:
- https://microbit-micropython.readthedocs.io/en/v2-docs/pin.html
- https://microbit-micropython.readthedocs.io/en/v2-docs/spi.html
